tests/test_mdarray_ctors.cpp

```diff
@@ -143,15 +143,33 @@ void check_correctness(MDA& m, size_t rank, size_t rank_dynamic,
   ASSERT_EQ(m.rank_dynamic(), rank_dynamic);
   if(rank>0) {
     ASSERT_EQ(m.extent(0), extent_0);
-    ASSERT_EQ(m.stride(0), stride_0);
+#if MDSPAN_HAS_CXX_20
```

Comment:
can't we just move that above 144. After all runtime wise these checks anyway only execute if rank>0 and we don't need to check that it compiles, there are other tests which do taht.

Reply:
And I mean with that we only need one big constexpr if protection for all runtime check rank clauses.

Reply:
That makes sense.  Updated.